pkg/operator/controller/clientca-configmap/controller.go

```diff
@@ -191,7 +191,7 @@ func (r *reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 	}
 
 	const finalizer = "ingresscontroller.operator.openshift.io/finalizer-clientca-configmap"
-	if len(ic.Spec.ClientTLS.ClientCA.Name) != 0 && !slice.ContainsString(ic.Finalizers, finalizer) {
+	if len(ic.Spec.ClientTLS.ClientCA.Name) != 0 && ic.DeletionTimestamp == nil && !slice.ContainsString(ic.Finalizers, finalizer) {
```

Comment:
Could you add a unit test for this change?

Reply:
https://github.com/openshift/cluster-ingress-operator/compare/d6ec5434cf35d6046264ceb5d652a5cc22cae4f3..b3ea6755c73f9e0a0a4980ae2deb838e072fda95 adds a unit test.